File: Programming Basics/Exam/deerOfSanta.js
function deerOfSanta(input) {

    let numberOfDaysForVacation = Number(input[0]);
    let foodInKG = Number(input[1]);
    let foodPerDayForTheFirstDeer= Number(input[2]);
    let foodPerDayForTheSecondDeer= Number(input[3]);
    let foodPerDayForTheThirdDeer= Number(input[4]);

    let foodNeededForTheFirstDeer = numberOfDaysForVacation * foodPerDayForTheFirstDeer;
    let foodNeededForTheSecondDeer = numberOfDaysForVacation * foodPerDayForTheSecondDeer;
    let foodNeededForTheThirdDeer = numberOfDaysForVacation * foodPerDayForTheThirdDeer;

    let foodNeeded = foodNeededForTheFirstDeer +foodNeededForTheSecondDeer + foodNeededForTheThirdDeer;
    
    let kg = Math.abs(foodInKG - foodNeeded);

    if(foodInKG >= foodNeeded) {
        console.log(`${Math.floor(kg)} kilos of food left.`);
    } else {
        console.log(`${Math.ceil(kg)} more kilos of food are needed.`);
    }

}

deerOfSanta(["5",
"10",
"2.1",
"0.8",
"11"])

File: Programming Basics Java/src/Exam/DeerOfSanta.java
package Exam;

import java.util.Scanner;

public class DeerOfSanta {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int missingDays = Integer.parseInt(scanner.nextLine());
        int food = Integer.parseInt(scanner.nextLine());
        double foodPerDayForFirstDeer = Double.parseDouble(scanner.nextLine());
        double foodPerDayForSecondDeer = Double.parseDouble(scanner.nextLine());
        double foodPerDayForThirdDeer = Double.parseDouble(scanner.nextLine());

        double totalFoodNeeded = (missingDays * foodPerDayForFirstDeer) + (missingDays * foodPerDayForSecondDeer) + (missingDays * foodPerDayForThirdDeer);

        double diff = Math.abs(food - totalFoodNeeded);

        if (food >= totalFoodNeeded) {
            System.out.printf("%d kilos of food left.", (int) Math.floor(diff));
        } else {
            System.out.printf("%d more kilos of food are needed.", (int) Math.ceil(diff));
        }
    }
}
